### ISSUE TITLE
Translation keys get deleted

### ISSUE THREAD

**Issue by TheFehr** [ctf0/Lingo#11]
I do not yet know when exactly and why.

But I had it multiple times that Lingo deleted a translation key.
One time it was exactly the translations string that I last edited. But it also (randomly) deletes others.

**Comment by TheFehr** [comment]
```
fetch("http://localhost/dev/lingo/save-file-data", {
  "body": "{
    \"file_name\":\"platform/roles.php\",
    \"dir_name\":null,
    \"data\":[
      {\"name\":\"actions.create.name\",\"locales\":{\"de\":null,\"en\":null}},
      {\"name\":\"actions.create.success\",\"locales\":{\"de\":null,\"en\":null}},
      {\"name\":\"actions.delete.name\",\"locales\":{\"de\":null,\"en\":null}},
      {\"name\":\"actions.delete.success\",\"locales\":{\"de\":null,\"en\":null}},
      {\"name\":\"actions.restore.name\",\"locales\":{\"de\":null,\"en\":null}},
      {\"name\":\"actions.restore.success\",\"locales\":{\"de\":null,\"en\":null}},
      {\"name\":\"actions.save.name\",\"locales\":{\"de\":null,\"en\":null}},
      {\"name\":\"actions.static.no_resource\",\"locales\":{\"de\":null,\"en\":null}},
      {\"name\":\"actions.update.name\",\"locales\":{\"de\":null,\"en\":null}},
      {\"name\":\"actions.update.success\",\"locales\":{\"de\":null,\"en\":null}},
      {\"name\":\"definition.name\",\"locales\":{\"de\":null,\"en\":null}},
      {\"name\":\"navigation.title\",\"locales\":{\"de\":null,\"en\":null}},
      {\"name\":\"screens.create.breadcrumb\",\"locales\":{\"de\":null,\"en\":null}},
      {\"name\":\"screens.description\",\"locales\":{\"de\":null,\"en\":null}},
      {\"name\":\"screens.edit.breadcrumb\",\"locales\":{\"de\":null,\"en\":null}},
      {\"name\":\"screens.edit.description\",\"locales\":{\"de\":\"Definiere welche Berechtigungen eine Rolle zusammenfasst\",\"en\":\"Define what permissions this role has\"}},
      {\"name\":\"screens.edit.name\",\"locales\":{\"de\":\"Rolle bearbeiten\",\"en\":\"Edit role\"}},
      {\"name\":\"screens.list.actions.edit\",\"locales\":{\"de\":\"Bearbeiten\",\"en\":\"Edit\"}},
      {\"name\":\"screens.list.actions.view\",\"locales\":{\"de\":null,\"en\":null}},
      {\"name\":\"screens.list.breadcrumb\",\"locales\":{\"de\":\"Alle Rollen\",\"en\":\"All roles\"}},
      {\"name\":\"screens.list.description\",\"locales\":{\"de\":\"Rollen definieren eine Sammlung von Zugangsrechten\",\"en\":\"Roles define a collection of permissions\"}},
      {\"name\":\"screens.list.name\",\"locales\":{\"de\":\"Rollen\",\"en\":\"Roles\"}},
      {\"name\":\"screens.view.actions.edit\",\"locales\":{\"de\":null,\"en\":null}},
      {\"name\":\"screens.view.fields.created_at.title\",\"locales\":{\"de\":null,\"en\":null}},
      {\"name\":\"screens.view.fields.id.title\",\"locales\":{\"de\":null,\"en\":null}},
      {\"name\":\"screens.view.fields.name.title\",\"locales\":{\"de\":null,\"en\":null}},
      {\"name\":\"screens.view.fields.slug.title\",\"locales\":{\"de\":null,\"en\":null}}
    ]
  }",
  "method": "POST",
  "mode": "cors",
  "credentials": "include"
});
```

This request deleted those keys:
platform/roles.screens.list.actions.edit
platform/roles.screens.description
platform/roles.screens.view.fields.name.title
platform/roles.screens.view.fields.created_at.title
platform/roles.screens.view.fields.id.title
platform/roles.screens.view.fields.name.title

**Comment by ctf0** [comment]
this is probably related to not supporting more than 3 levels deep, which will require a better recursive implementation than the current one.

**Comment by TheFehr** [comment]
Ah, I didn't really look into your code yet.
But that would clearly explain the issue.
May I ask why you split the translations into separate arrays?

I built a console command to ensure a certain list of translation keys that are defined by annotations and certain inheritances in my project. And I saved the translations as ``'a.b.c.d' => 'translation'`` and you seem to save it as ``'a' => [ 'b' => [ 'c' => [ 'd' => 'translation' ] ] ]``
